libs/layer-composer/src/layer-composer.ts

```diff
@@ -81,25 +79,22 @@ export class LayerComposer {
     newGeneratorStyles.layers = newGeneratorStyles.layers.map((layer) => {
       const newLayer = { ...layer }
       if (!newLayer.layout) {
-        newLayer.layout = {} as AnyLayout
+        newLayer.layout = {} as LayerSpecification['layout']
       }
       newLayer.layout = {
         ...newLayer.layout,
         visibility: isConfigVisible(generatorConfig),
       }
       if (!newLayer.paint) {
-        newLayer.paint = {} as AnyPaint
+        newLayer.paint = {} as LayerSpecification['paint']
       }
       if (!newLayer.metadata) {
-        newLayer.metadata = {
-          generatorId: generatorConfig.id,
-          generatorType: generatorConfig.type as GeneratorType,
-        }
-      } else {
-        newLayer.metadata = {
-          ...newLayer.metadata,
-          generatorType: generatorConfig.type as GeneratorType,
-        }
+        newLayer.metadata = {}
+      }
+      newLayer.metadata = {
+        ...newLayer.metadata,
+        generatorId: generatorConfig.id,
+        generatorType: generatorConfig.type as GeneratorType,
```

Comment:
> the previous version was not adding the generatorId to the metadata when the layer received as argument had metadata. Is this correct?

yes it is, but I did the change for simplicity and consistency to apply the common configuration to each generator because:
 
- we ensure every layer will have a `generatorId` 
- we can avoid repeating [this](https://github.com/GlobalFishingWatch/frontend/blob/maplibre/libs/layer-composer/src/generators/context/context.ts#L107) in every generator

Reply:
Perfect then, it makes sense.